pkg/util/resize/configspec.go

```diff
@@ -482,6 +483,41 @@ func compareNPIV(
 	}
 }
 
+// compareVirtualNuma compares the virtual NUMA configurations set in the config spec.
+func compareVirtualNuma(
+	ci vimtypes.VirtualMachineConfigInfo,
+	cs vimtypes.VirtualMachineConfigSpec,
+	outCS *vimtypes.VirtualMachineConfigSpec) {
+	if cs.VirtualNuma == nil {
+		return
+	}
+
+	if ci.NumaInfo == nil {
+		outCS.VirtualNuma = &vimtypes.VirtualMachineVirtualNuma{
+			CoresPerNumaNode:       cs.VirtualNuma.CoresPerNumaNode,
```

Comment:
If the configInfo.NumInfo was Nil earlier - No Numa node setting was specified on the VM until now. The setting
`cs.VirtualNuma.{CoresPerNumaNode: 0, ExposeVnumaOnCpuHotadd: nil} ` is a valid setting since this `CoresPerNumaNode: 0` means use automatic Numa node sizing. 

In the case where configInfo.NumInfo was non-nil with a `non-zero CoresPerNumaNode` -- again the `CoresPerNumaNode: 0` is valid and it clears out manual overrides and makes the resizing of numa nodes automatic.